packages/tabs/src/Tabs.tsx

```diff
@@ -163,105 +173,116 @@ function Tabs({
     ? setControlledSelected
     : setUncontrolledSelected;
 
-  function handleChange(
-    e: React.SyntheticEvent<Element, MouseEvent>,
-    index: number,
-  ) {
-    setSelected(index);
-  }
+  const handleChange = useCallback(
+    (e: React.SyntheticEvent<Element, MouseEvent>, index: number) => {
+      setSelected(index);
+    },
+    [setSelected],
+  );
 
-  const getEnabledIndexes: () => [Array<number>, number] = () => {
+  const getEnabledIndexes: () => [Array<number>, number] = useCallback(() => {
     const enabledIndexes = childrenArray
       .filter(child => !child.props.disabled)
       .map(child => childrenArray.indexOf(child));
 
     return [enabledIndexes, enabledIndexes.indexOf(selected!)];
-  };
-
-  const handleArrowKeyPress = (e: KeyboardEvent) => {
-    if (!(e.metaKey || e.ctrlKey)) {
-      if (e.keyCode === keyMap.ArrowRight) {
-        const [enabledIndexes, current] = getEnabledIndexes();
-        setSelected(enabledIndexes[(current + 1) % enabledIndexes.length]);
-      } else if (e.keyCode === keyMap.ArrowLeft) {
-        const [enabledIndexes, current] = getEnabledIndexes();
-        setSelected(
-          enabledIndexes[
-            (current - 1 + enabledIndexes.length) % enabledIndexes.length
-          ],
-        );
+  }, [childrenArray, selected]);
+
+  const handleArrowKeyPress = useCallback(
+    (e: KeyboardEvent) => {
+      if (!(e.metaKey || e.ctrlKey)) {
+        if (e.keyCode === keyMap.ArrowRight) {
+          const [enabledIndexes, current] = getEnabledIndexes();
+          setSelected(enabledIndexes[(current + 1) % enabledIndexes.length]);
+        } else if (e.keyCode === keyMap.ArrowLeft) {
+          const [enabledIndexes, current] = getEnabledIndexes();
+          setSelected(
+            enabledIndexes[
+              (current - 1 + enabledIndexes.length) % enabledIndexes.length
+            ],
+          );
+        }
       }
-    }
-  };
+    },
+    [getEnabledIndexes, setSelected],
+  );
 
-  const tabs = React.Children.map(children, (child, index) => {
-    if (!isComponentType<'Tab'>(child, 'Tab')) {
-      return child;
-    }
+  const { tabs, tabpanels } = useMemo(() => {
+    const tabs: Array<React.ReactElement> = [];
 
-    return React.cloneElement(child, {
-      key: index,
-      ariaControl: `tab-${index}`,
-      selected: selected === index,
+    const tabpanels = React.Children.map(children, (child, index) => {
+      if (!isComponentType<'Tab'>(child, 'Tab')) {
```

Comment:
It would be better to remove `<'Tab'>` altogether, since it expects a `ReactElement` not a string

Reply:
👍